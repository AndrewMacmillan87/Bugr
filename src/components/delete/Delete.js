import React from "react";

import PropTypes from "prop-types";

import "./delete.css";

export const Delete = ({ title, extraClasses, revealFunc, onClick }) => {
    const deleteClasses = extraClasses
        ? `delete-project-container ${extraClasses}`
        : "delete-project-container";

    return (
        <div className={deleteClasses}>
            <p className='delete-project-msg'>Delete project?</p>
            <div className='delete-project-btn-container'>
                <button className='delete-project-btn'>
                    <i
                        className='do-delete fas fa-check-circle'
                        onClick={() => {
                            onClick();
                            revealFunc("");
                        }}></i>
                </button>
                <button className='delete-project-btn'>
                    <i
                        className='do-not-delete fas fa-times-circle'
                        onClick={() => revealFunc("")}></i>
                </button>
            </div>
        </div>
    );
};

Delete.propTypes = {
    title: PropTypes.string,
    extraClasses: PropTypes.string,
    revealFunc: PropTypes.func,
    onClick: PropTypes.func,
};

export default Delete;
