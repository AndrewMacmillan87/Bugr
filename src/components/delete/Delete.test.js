import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import Delete from "./Delete";

describe("Delete component", () => {
    test("Renders without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(<Delete />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it("renders correctly", () => {
        const tree = renderer.create(<Delete />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe("Delete HTML elements", () => {
        it("should render 2 dev elements", () => {
            const wrapper = mount(<Delete />);
            expect(wrapper.find("div").length).toEqual(2);
        });

        it("should render a p element", () => {
            const wrapper = mount(<Delete />);
            expect(wrapper.find("p").length).toEqual(1);
        });

        it("should render 2 button elements", () => {
            const wrapper = mount(<Delete />);
            expect(wrapper.find("button").length).toEqual(2);
        });

        it("should render 2 i elements", () => {
            const wrapper = mount(<Delete />);
            expect(wrapper.find("i").length).toEqual(2);
        });
    });
});
