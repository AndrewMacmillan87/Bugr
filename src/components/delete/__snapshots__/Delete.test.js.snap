// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidePanel component renders correctly 1`] = `
<div
  className="delete-project-container"
>
  <p
    className="delete-project-msg"
  >
    Delete 
    ?
  </p>
  <div
    className="delete-project-btn-container"
  >
    <button
      className="delete-project-btn"
    >
      <i
        className="do-delete fas fa-check-circle"
      />
    </button>
    <button
      className="delete-project-btn"
    >
      <i
        className="do-not-delete fas fa-times-circle"
        onClick={[Function]}
      />
    </button>
  </div>
</div>
`;
